intelmq/bots/experts/idea/expert.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+from collections import Sequence, Mapping
+from base64 import b64decode
+from uuid import uuid4
+from urllib.parse import quote_plus
+
+from intelmq.lib.bot import Bot
+from intelmq.lib.message import Event
+
+
+def quot(s):
+    return quote_plus(s or "")
+
+def addr4(s):
+    return s if ":" not in s else None
+
+def addr6(s):
+    return s if ":" in s else None
+
+
+class IdeaExpertBot(Bot):
+
+    type_to_category = {
+        "phishing": "Fraud.Phishing",
+        "ddos": "Availability.DDoS",
+        "spam": "Abusive.Spam",
+        "scanner": "Recon.Scanning",
+        "dropzone": "Information.UnauthorizedAccess",
+        "malware": "Malware",
+        "botnet drone": "Malware",
+        "ransomware": "Malware",
+        "malware configuration": "Malware",
+        "c&c": "Intrusion.Botnet",
+        "exploit": "Attempt.Exploit",
+        "brute-force": "Attempt.Login",
+        "ids alert": "Attempt.Exploit",
+        "defacement": "Intrusion.AppCompromise",
+        "compromised": "Intrusion.AdminCompromise",
+        "backdoor": "Intrusion.AdminCompromise",
+        "vulnerable service": "Vulnerable.Open",
+        "blacklist": "Other",
+        "unknown": "Other",
+        "test": "Test"
+    }
+
+    type_to_source_type = {
+        # Added nonstandard Dropzone, MalwareConf, will consider adding to Idea spec
+
+        "phishing": "Phishing",
+        "dropzone": "Dropzone",
+        "botnet drone": "Botnet",
+        "malware configuration": "MalwareConf",
+        "c&c": "CC"
+    }
+
+    def _init_translation(self):
```

Comment:
Ahha, I got misdirected by markdown reformatting of `__init__` and thought we're talking the same thing. Now I see your point, the Bot has the `init` method, which gets called during `__init__`. Ok, makes sense.